src/main/java/view/ListView.java

```diff
@@ -218,18 +221,30 @@ private void addFilteredItem(int index) {
     /**
      * Removes the selected movie from the list.
      */
-    private void removeSelectedMovie() {
-        for (int i = 0; i < radioButtonList.size(); i++) {
-            if (radioButtonList.get(i).isSelected()) {
-                radioButtonList.remove(i);
-                movieDescriptions.remove(i);
-                mediaListTable.remove(i);
-                break;
+    void removeSelectedMovie() {
+        final int selectedRow = mediaListTable.getSelectedRow();
+        if (selectedRow != -1) {
+            final ListState state = listViewModel.getState();
+            final String collectionName = state.getCurrentCollectionName();
+
+            // Check if the collection name is null
+            if (collectionName == null) {
+                System.err.println("Collection name is null. Cannot remove movie.");
+            }
+            else {
+                final Object value = mediaListTable.getValueAt(selectedRow, 0);
```

Comment:
Ooooh, nice catch here! Using the JTable's `getValueAt` method instead of the TableModel's method makes sure that the media name you get is accurate even if the user filters or sorts the table. I wouldn't have thought of that.
This is just a comment, but idk how to mark this as resolved, haha.